### ISSUE TITLE
Why you have 6 labels in the dataset?

### ISSUE THREAD

**Issue by xingchenzhang** [UttaranB127/STEP#3]
Hi,

Thank you very much for sharing your code. 

I am wondering why you have 6 labels in the dataset but in the paper 4 emotions are used.

Thanks!
Xingchen

**Comment by UttaranB127** [comment]
Hi Xingchen,

We originally had two extra labels, somewhat happy and somewhat angry. But in the final paper, we merged those with happy and angry respectively. We have kept all the six labels in the dataset in case future users find them useful.

**Comment by xingchenzhang** [comment]
Hi,

Thank you very much for your quick reply! Now I got it.

Actually I still have another questions about your code. In your code paper you mentioned T is 75 (time steps). But when I saw your codes and datasets, it seems that 18 time steps were used. I am not sure if I understood correctly.

In addition, could you also please help to give some explanation about how you build the adjacent matrix A? I am wondering how you express the connections between joints. 

Sorry for so many questions, and many thanks again for your nice work!

Bests,
Xingchen

**Comment by UttaranB127** [comment]
Hi,

Thanks a lot for your interest! 75 is the maximum number of time steps we have for a gait, i.e., for any individual gait, the number of time steps we have is less than or equal to 75. So it is quite possible that the gaits you are looking at have less than 75 time steps. For gaits with less than 75 time steps, we implemented periodic padding: we repeated those gaits till they have 75 or more time steps, then clipped the time steps to 75.

To create the adjacency matrix, we simply followed the human joint structure. The following are our 15 edges:
root - spine
spine - neck
neck - head
neck - left shoulder
left shoulder - left elbow
left elbow - left hand
neck - right shoulder
right shoulder - right elbow
right elbow - right hand
root - left hip
left hip - left knee
left knee - left foot
root - right hip
right hip - right knee
right knee - right foot
